src/components/form/form.js

```diff
@@ -381,6 +434,25 @@ class Form extends React.Component {
     }
   }
 
+  addSavePromptListener = () => {
+    this._window.addEventListener('beforeunload', this.checkIsFormDirty);
```

Comment:
what happens if there are multiple forms on the page?

Reply:
I just attempted this locally with 2 forms being present on a page, one with savePrompt enabled and one with it disabled. The pop-up only displayed when the first form was changed, and didn't when the latter was changed, which is what I'd expect.
In the case of 2 forms both with savePrompt enabled, editing both forms then attempting to navigate away from the page brought up the prompt once, wether choosing to stay on the page or to navigate away from the page